tsl/src/compression/algorithms/uuid_compress.c

```diff
@@ -455,6 +457,19 @@ uuid_compressed_recv(StringInfo buffer)
 	uint32 timestamp_size = pq_getmsgint32(buffer);
 	uint32 rand_b_and_variant_size = pq_getmsgint32(buffer);
 
+	CheckCompressedData(timestamp_size > 24);
+	CheckCompressedData(rand_b_and_variant_size > 0);
+	CheckCompressedData(rand_b_and_variant_size % sizeof(uint64) == 0);
+	CheckCompressedData(timestamp_size % sizeof(uint64) == 0);
+	CheckCompressedData(rand_b_and_variant_size / sizeof(uint64) < GLOBAL_MAX_ROWS_PER_COMPRESSION);
+
+	/* Maximum number of rows in the timestamp data with the worst possible compression */
+	uint32 timestamp_row_limit = (uint32) ((double) (timestamp_size - 24) * 8.0 / 68.0);
+	CheckCompressedData(timestamp_row_limit < GLOBAL_MAX_ROWS_PER_COMPRESSION);
```

Comment:
You don't have to check it separately if it won't cause OOB errors. Looks a little complicated, I'd remove if possible.